definitions/checks/check_for_newer_packages.rb

```diff
@@ -12,15 +12,54 @@ class Checks::CheckForNewerPackages < ForemanMaintain::Check
   end
 
   def run
-    exit_status, = ForemanMaintain.package_manager.check_update(packages: @packages,
-                                                                with_status: true)
-    assert(exit_status == 0, 'An update is available for one of these packages: '\
-                "#{@packages.join(',')}. Please update before proceeding.")
+    check_for_package_update
     if @manual_confirmation_version
       question = 'Confirm that you are running the latest minor release of Satellite '\
                 "#{@manual_confirmation_version}"
       answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
       abort! if answer != :yes
     end
   end
+
+  def check_for_package_update
+    exit_status, output = ForemanMaintain.package_manager.check_update(packages: @packages,
+                                                                       with_status: true)
+    packages_with_updates = []
+    if exit_status == 100
+      packages_with_updates = compare_pkg_versions(output).select do |_, result|
+        result == -1
+      end.keys
+    end
+
+    unless packages_with_updates.empty?
+      failure_message = 'An update is available for package(s): '\
+                        "#{packages_with_updates.join(',')}. Please update before proceeding!"
+      fail! failure_message
+    end
+  end
+
+  def packages_and_versions(output)
+    pkgs_versions = {}
+    pkg_details = output.split("\n\n")[1].split
+    iterator = 0
+    while iterator < 6
```

Comment:
I don't really understand this iterator and the magic numbers 6 and 3

Reply:
The check is for two packages `yum check-update` command,
1. We collect the packages and versions when exit code is 100 which is when updates available.
2. So at max it has 6 entries in array as below. Though I expect exact entries, better I should use pkg_details.length, 
~~~
['package1',
'version',
'repo',
'package1',
'version',
'repo']
~~~